datafusion/functions-aggregate/src/macros.rs

```diff
@@ -48,49 +48,98 @@ macro_rules! make_udaf_expr_and_func {
                 None,
             ))
         }
+
+        create_builder!(
+            $EXPR_FN,
+            $($arg)*,
+            $DOC,
+            $AGGREGATE_UDF_FN
+        );
+
         create_func!($UDAF, $AGGREGATE_UDF_FN);
     };
-    ($UDAF:ty, $EXPR_FN:ident, $($arg:ident)*, $distinct:ident, $DOC:expr, $AGGREGATE_UDF_FN:ident) => {
```

Comment:
I plan to introduce another macro for distinct, so we have `count_distinct`, `count_distinct_builder` expr::fn.

I change to `order_by` because `first_value` needs it. Also, I change the expression for `first_value` to single expression, since it does not expect variadic args.

Reply:
Instead of `count_distinct_builder` what do you think about adding a `distinct` type method instead?

Like
```rust
let agg = count_builder()
  .args(col("a"))
  .distinct()
  .build()?
```

🤔